### DISCUSSION TITLE
Connection to Codespace is unbearably slow

### DISCUSSION THREAD

**Discussion by brawaru** [community/community#34864] [Codespaces]
It seems that connections to Codespaces are throttled to unbearable speeds. I have to debug JavaScript app and in development builds code chunks can reach up to 5 MB in size, which is nothing for my network connection, yet it takes 2 minutes (!!!) to load through bound port of Codespace on `localhost`.

![Screenshot of DevTools showing files downloaded through network, for the ‘app.js’ that is of size 2.4 MB, it took 42 seconds to download it; for the second ‘app.js’ of size 4.1 MB it took 2.1 minutes; third ‘app.js’ of 2.9 MB took 2 minutes; and ‘index.js’ of size 186 kB took 1.5 minutes to download](https://user-images.githubusercontent.com/10401817/193453550-898cb330-4490-459c-a97c-868ee30350b2.png)

Codespaces competitor, Gitpod, doesn't seem to have this problem, from there this chunk will be downloaded before you'll snap your fingers. Sadly, they have other problems to be a viable alternative.

And even worse, while you're busy loading this chunk of JavaScript in the browser, forget about being able to save / open any files or folders in VS Code itself, the channel is too overloaded for VS Code to be able to do anything. At the end it may also complain about the ‘network delay’:

![Screenshot of VS Code warning from Codespaces extensions about network delay of 32,594 ms](https://user-images.githubusercontent.com/10401817/193453145-1bbf3c85-725e-48e5-93ac-18691e615ad1.png)

I even tried to connect with VPN from various other European countries, hoping that maybe it's my network problem, but there are zero changes expect for the terminal delay, so I doubt that's my issue. Don't think that's app's issue either, it uses less that GB of RAM and rarely reaches more than 10% CPU on any of the cores, and once again, it worked perfectly fine on Gitpod.

I really hope GitHub will address this. That's just ridiculous for the dev experience and kills a genuinely good service.

**Comment by joshspicer** [comment]
Hey @Brawaru!  Codespaces is geographically located in four Azure regions - would you mind double checking that you aren't accidentally pinning yourself to a distant region: https://docs.github.com/en/codespaces/customizing-your-codespace/setting-your-default-region-for-github-codespaces.

We haven't heard similar reports, but will look into latency issues with the information you've provided

**Reply by sheuvembo** [reply]
hello @joshspicer I live in africa and have been trying to work on my problem sets but i cant seem to find a region thats fast enough to load the codespaces.
They all take a minimum of an hour to load.
Please help!

**Reply by joshspicer** [reply]
Apologies for the lack of regions close to your physical location. We have slowly been moving into new geographies, unfortunately it is not a trivial task and I don't have any concrete updates for you yet on where or where we'd be moving into next.

We'll post to https://github.blog/changelog/ if there's any updates to share

**Comment by Israeltheminer** [comment]
Did you ever find a solution to this? @brawaru  @sheuvembo

**Reply by brawaru** [reply]
I mean if you can count it as a solution, I don't use Codespaces anymore :)

**Reply by Israeltheminer** [reply]
Also had to switch to dev containers

**Comment by kleenkanteen** [comment]
I used codesapces for 30 hours the past 2 weeks. A few times a day it would get really slow so I couldn't search for and load files quickly, or anything really. Every few days it would try to reconnect indefinitely and I'd have to stop my current codespace and restart it for it to work. Stopped using it.

**Reply by ukpagrace** [reply]
Codespaces startup time and load time is so slow it takes up to 30mins to load and Upon that it disconnects after 15 mins.

Having to do this again, when it connects is hell

**Reply by 825i** [reply]
I am having the exact same issue. Some codespaces seem to take forever to build even the simplest dev-container. I think it's likely congestion and being pushed to a runner (which creates the codespace for me) that has a high queue. So once it's simply "my turn" everything happens quick enough but there's a hidden queue time I have to wait until everything actually begins to be deployed.

**Comment by dcsturman** [comment]
I’ve been trying to use Codespaces off an iPad while on the Shinkansen in Japan (and my codespace is the SE Asia region).  While my internet is not great but solid (consistent 2-5Mbps on speed tests) Codespaces is very slow. Or rather, periodically has to reload which takes ~5-10 min each time.

Has anyone found an alternative that performs a bit better? 

 A (possibly obnoxiously unsolicited) suggestion for the Codespaces team - if you have good laptop or workstations, just get rid of them.  Get a very weak device and possibly find a way to throttle your network, and code every day off of that. I’ve used this technique for teams I’ve lead where users are on a range of devices. It’s painful at first but you get a much better product in the end.

**Reply by arashatt** [reply]
I think the are some throttling is happening from GitHub. I am in Iran and even with a good connection, I cannot develop any code. I hope that there is some explanation to the problem.

**Comment by sgt-tachyon** [comment]
i have more than 40 activities to do using codespace by the end of my term. these short activities would have been trivial if it wasn't for the abysmal connection. 5 minute assignments turn into 30 minutes each of retrying the code checker to update my grade.
